q2_american_gut/_fetch.py

```diff
@@ -0,0 +1,205 @@
+        test_ctx = context.lower()
+
+        # american gut amplicon data are only 16S v4
+        if '16s' not in test_ctx or 'v4' not in test_ctx:
+            continue
+
+        # TODO: the release candidate version of the qiita redbiom database
+        # has updated context names which include the reference used.
+        # if we're closed reference, restrict to greengenes as a reference
+        #if processing_type != 'deblur' and 'greengenes' not in test_ctx:
+        #    continue
+
+        if processing_type in test_ctx and "%dnt" % trim_length in test_ctx:
+            found = context
+
+    if found is None:
+        msg = "Cannot find %s-%dnt for 16S data" % (processing_type, trim_length)
+        raise ValueError(msg)
+
+    return found
+
+
+def _get_featuredata_from_table(table):
+    """Extract the observations and interpret as skbio.DNA"""
+    if table.is_empty():
+        raise ValueError("No features")
+
+    return DNAIterator((skbio.DNA(i, metadata={'id': i})
+                                  for i in table.ids(axis='observation')))
+
+
+def _fetch_taxonomy(processing_type, table, threads):
+    """Fetch taxonomy based on the processing type
+
+    Parameters
+    ----------
+    processing_type : {'deblur', 'closed-reference'}
+        The processing type which determines how we obtain taxonimic detail.
+    table : biom.Table
+        The FrequencyTable
+    threads : int
+        The number of threads to use
+
+    Returns
+    -------
+    FeatureData[Taxonomy]
+    """
+    if processing_type == 'deblur':
+        from qiime2.plugins import feature_classifier
+        dna_iter = _get_featuredata_from_table(table)
+
+        reads = qiime2.Artifact.import_data('FeatureData[Sequence]', dna_iter)
+        classifier = qiime2.Artifact.load(resource_filename(*CLASSIFIER))
+        tax, = feature_classifier.methods.classify_sklearn(reads,
+                                                           classifier,
+                                                           n_jobs=threads)
+    else:
+        tax_table = pd.DataFrame([(i, m['taxonomy'], 1.0)
+                                  for v, i, m in table.iter(axis='observation',
+                                                            dense=False)],
+                                 columns=['Feature ID', 'Taxon', 'Confidence'])
+        tax_table.set_index('Feature ID', inplace=True)
+        tax = qiime2.Artifact.import_data('FeatureData[Taxonomy]', tax_table)
+
+    return tax
+
+
+def _fetch_phylogeny(processing_type, table, threads, debug):
+    """Fetch phylogeny based on the processing type
+
+    Parameters
+    ----------
+    processing_type : {'deblur', 'closed-reference'}
+        The processing type which determines how we resolve a phylogeny.
+    table : biom.Table
+        The FrequencyTable
+    threads : int
+        The number of threads to use
+    debug : bool, optional
+        If debug, use a small small tree for insertion to avoid spinup
```

Comment:
small small -> small? ... unless the tree is indeed small small 😆

Reply:
There was a small tree and a tiny tree, so I figure small small is right